### PR TITLE
thumb styling fix #55

### LINKED ISSUE #55: Add support for the hover style option
I need set a custom behavior when the user pass the mouse hover or focus in the thumb change the style, but custom style don't support this functionality.

Example:

https://codesandbox.io/s/objective-sound-o4s5x

### PR DESCRIPTION
**Pull request by wangzuo**

### CODE REVIEW

**Comment by RodolfoSilva** [comment]
Much better! @wangzuo 😜

File: src/slider.js [hunk 1]
```diff
@@ -155,9 +155,18 @@ const Slider = ({
   if (xreverse) valueStyle.left = 100 - pos.left + '%';
   if (yreverse) valueStyle.top = 100 - pos.top + '%';
 
-  const thumbStyle = {};
-  thumbStyle.left = xreverse ? 100 - pos.left + '%' : pos.left + '%';
-  thumbStyle.top = yreverse ? 100 - pos.top + '%' : pos.top + '%';
+  const handleStyle = {
```
Comment (RodolfoSilva) [inline src/slider.js:158]: Another approach without mutation is use a simplified way. 
```javascript
const handleStyle = {
  position: 'absolute',
  transform: 'translate(-50%, -50%)',
  left: `${axis === 'x' ? 50 : (xreverse ? 100 - pos.left : pos.left)}%`,
  top: `${axis === 'y' ? 50 : (xreverse ? 100 - pos.top : pos.top)}%`,
};
```

File: src/slider.js [hunk 2]
```diff
@@ -191,15 +185,16 @@ const Slider = ({
...
+        style={handleStyle}
         onTouchStart={handleMouseDown}
         onMouseDown={handleMouseDown}
         onClick={function(e) {
           e.stopPropagation();
           e.nativeEvent.stopImmediatePropagation();
         }}
-      />
+      >
+        <div css={styles.thumb} />
```
Comment (RodolfoSilva) [inline src/slider.js:196]: With this approach we can provide a way to user pass the **thumb** as a custom component too.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## [Unreleased]
 
+- improve styling for thumb
+- fix typescript definition
+- thumb with pointer cursor by default
```
Comment (RodolfoSilva) [inline CHANGELOG.md:7]: Nice!

File: src/slider.js (hunk 2 above)
Comment (wangzuo) [reply src/slider.js:196]: yep, probably add that later

**Pull request merged**